src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -356,9 +344,21 @@ export default {
 		},
 
 		/**
-		 * WebImage handlers
+		 * Gets the list of supported social networks that provide avatar download
+		 *
+		 * @returns {array} array of supported social networks
+		 */
+		async getSupportedNetwoks() {
+			const result = await axios.get(generateUrl('/apps/contacts/api/v1/social/supported/avatar'))
+			if (result.status === 200) {
+				this.supportedNetworks = result.data
+			}
+		},
```

Comment:
Hey, please use an initial state for this, no need for an extra request :wink: 
https://github.com/nextcloud/contacts/blob/24581f0b0ce8a89261d164fcee5421cf36b88c12/lib/Controller/PageController.php#L69
https://github.com/nextcloud/contacts/blob/24581f0b0ce8a89261d164fcee5421cf36b88c12/src/models/rfcProps.js#L23-L30

Reply:
Much better! 
Was it correct to add the load-state to the PageController?

Reply:
Yes :)